clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -500,9 +506,9 @@ private KafkaAdminClient(AdminClientConfig config,
                              KafkaClient client,
                              TimeoutProcessorFactory timeoutProcessorFactory,
                              LogContext logContext) {
-        this.defaultTimeoutMs = config.getInt(AdminClientConfig.REQUEST_TIMEOUT_MS_CONFIG);
         this.clientId = clientId;
         this.log = logContext.logger(KafkaAdminClient.class);
+        configureDefaultApiTimeoutMsAndRequestTimeoutMs(config);
```

Comment:
Could we move this extra initialization logic to the factories and let the constructor accept the two new values? That we also allow us to make them final.